tests/unit/test_command_validation.py

```diff
@@ -37,3 +42,56 @@ async def test_trigger_invalid(flow_strs, expected_msg):
     with pytest.raises(InputError) as exc_info:
         flow_opts(flow_strs, False)
     assert str(exc_info.value) == expected_msg
+
+
+async def test_is_tasks(set_cycling_type):
+    set_cycling_type(ISO8601_CYCLING_TYPE)
+
+    # tokens should be parsed
+    assert is_tasks({'20000101T0000Z/a', '20010101T0000Z/b'}) == {
+        TaskTokens('20000101T0000Z', 'a'),
+        TaskTokens('20010101T0000Z', 'b'),
+    }
+
+    # cycle points should be standardised unless they are globs
+    assert is_tasks({'2000/a', '20010101/b', '*/c', '[23]000/d'}) == {
+        TaskTokens('20000101T0000Z', 'a'),
+        TaskTokens('20010101T0000Z', 'b'),
+        TaskTokens('*', 'c'),
+        TaskTokens('[23]000', 'd'),
+    }
+
+    # the namespace should default to "root" unless provided
+    assert is_tasks({'*', '2000'}) == {
+        TaskTokens('*', 'root'),
+        TaskTokens('20000101T0000Z', 'root'),
+    }
+
+    # invalid IDs result in errors
+    with pytest.raises(InputError, match='Invalid ID: //, ///, ////'):
+        is_tasks({'//', '///', '////'})
```

Comment:
Convince me that it _only_ identifies the invalid ones?
```suggestion
        is_tasks({'//', '///', '////', '2000/a'})
```